pkg/middleware/auth.go

```diff
@@ -220,3 +222,49 @@ func OrgAdminDashOrFolderAdminOrTeamAdmin(ss sqlstore.Store, ds dashboards.Dashb
 		accessForbidden(c)
 	}
 }
+
+func ReqRoleForAppRoute(ps plugins.Store, preCond web.Handler) func(c *models.ReqContext) {
+	return func(c *models.ReqContext) {
+		if f, ok := preCond.(func(c *models.ReqContext)); ok {
+			f(c)
+		}
+
+		pluginID := web.Params(c.Req)[":id"]
+
+		p, exists := ps.Plugin(c.Req.Context(), pluginID)
+		if !exists {
+			c.JsonApiErr(404, "Plugin not found", nil)
+			return
+		}
+
+		normalizePath := func(p string) string {
+			return filepath.Clean(strings.TrimPrefix(p, "/"))
+		}
+
+		found := false
+		allowed := false
+		path := normalizePath(web.Params(c.Req)["*"])
+		for _, i := range p.Includes {
+			if i.Type != plugins.PageIncludeType {
+				continue
+			}
+
+			if normalizePath(i.Path) == path {
+				found = true
+				if c.HasRole(i.Role) {
```

Comment:
Think if `i.Role` is not set we should also set `allowed = true`. WDYT?

Reply:
Yeah we probably should just in case [this](https://github.com/grafana/grafana/blob/main/pkg/plugins/manager/loader/loader.go#L227-L231) ever gets removed

Reply:
Ohh wasn't aware of that :smile: No biggie then, but agree with you that it's a bit fragile